module/VuFind/src/VuFind/View/Helper/Root/RecordDataFormatter.php

```diff
@@ -63,6 +63,68 @@ public function specSortCallback($a, $b)
         return ($a['pos'] ?? 0) <=> ($b['pos'] ?? 0);
     }
 
+    /**
+     * Should we allow a value? (Performs zero checking on a value that
+     * evaluates to empty).
+     *
+     * @param mixed $value   Data to check for zero value.
+     * @param array $options Rendering options.
+     *
+     * @return bool
+     */
+    protected function allowValue($value, $options)
+    {
+        if (!empty($value)) {
+            return true;
+        }
+        $allowZero = $options['allowZero'] ?? true;
+        return $allowZero && ($value === 0 || $value === '0');
+    }
+
+    /**
+     * Return rendered text (or null if nothing to render).
+     *
+     * @param RecordDriver $driver  Record driver object
+     * @param string       $field   Field being rendered (i.e. default label)
+     * @param mixed        $data    Data to render
+     * @param array        $options Rendering options
+     *
+     * @return array
+     */
+    protected function render($driver, $field, $data, $options)
+    {
+        // Check whether the data is worth rendering.
+        if (!$this->allowValue($data, $options)) {
+            return null;
+        }
+
+        // Determine the rendering method to use, and bail out if it's illegal:
+        $method = empty($options['renderType'])
+            ? 'renderSimple' : 'render' . $options['renderType'];
+        if (!is_callable([$this, $method])) {
+            return null;
+        }
+
+        // If the value evaluates false, we should double-check our zero handling:
+        $value = $this->$method($driver, $data, $options);
+        if (!$this->allowValue($value, $options)) {
+            return null;
+        }
+
+        // Special case: if we received an array rather than a string, we should
+        // return it as-is (it probably came from renderMulti()).
+        if (is_array($value)) {
```

Comment:
Should there be an allowValue-like filter function for the returned values, or is it expected that renderMulti never returns non-allowed values, e.g. empty strings?

Reply:
Because renderMulti wraps around the other existing render methods, I don't think it needs its own separate filtering logic. However, you are correct that I wasn't handling the existing return values properly, so when a different render method returned null, it was causing an error. I've made some revisions (and expanded the test class) to take care of this. Thanks!